File: go/7.go
func reverse_s(s string) string {
	chars := []rune(s)
	for i, j := 0, len(chars)-1; i < j; i, j = i+1, j-1 {
		chars[i], chars[j] = chars[j], chars[i]
	}
	return string(chars)
}

func reverse(x int) int {
    var ret int64
    if x > 0 {
        ret, _ = strconv.ParseInt(reverse_s(strconv.Itoa(x)), 10, 64)
    } else {
        ret, _ = strconv.ParseInt(reverse_s(strconv.Itoa(x)[1:]), 10, 64)
        ret = -ret
    }
    if ret > 0x7fffffff || ret <= -0x80000000 {
        return 0
    }
    return int(ret)
}

File: js/7.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function (x) {
    let ret = x > 0 ? +(x + '').split('').reverse().join('') : -(x + '').split('').splice(1).reverse().join('')
    if (ret > 0x7fffffff || ret <= -0x80000000) return 0
    return ret
};